assets/js/extensions/shipping-methods/pickup-location/edit-location/index.tsx

```diff
@@ -69,8 +69,12 @@ const EditLocation = ( {
 					<Button
 						variant="primary"
 						onClick={ () => {
-							onSave( values );
-							onClose();
+							const form =
+								formRef?.current as unknown as HTMLFormElement;
+							if ( form.reportValidity() ) {
+								onSave( values );
+								onClose();
+							}
```

Comment:
Button here is outside the form, so we use the ref instead of `event.target.form`